subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyReportTaskIntegrationTest.groovy

```diff
@@ -804,4 +804,36 @@ compile
 \\--- org.original:original:1.0 -> org.other:another:0.1
 """
     }
+
+    void "doesn't fail if a configuration is not resolvable"() {
+        mavenRepo.module("foo", "foo", '1.0').publish()
+        mavenRepo.module("foo", "bar", '2.0').publish()
+
+        file("build.gradle") << """
+            repositories {
+               maven { url "${mavenRepo.uri}" }
+            }
+            configurations {
+                api.canBeResolved = false
+                compile.extendsFrom api
+            }
+            dependencies {
+                api 'foo:foo:1.0'
+                compile 'foo:bar:2.0'
+            }
+        """
+
+        when:
+        run ":dependencies"
+
+        then:
+        output.contains """
+api
```

Comment:
`api` should probably be excluded from the output I would think, or perhaps represented differently to simply list the bucket of unresolved dependencies rather than the graph of resolved dependencies.

I think rendering the result of a copy of the configuration is potentially misleading, as nothing will ever use this result, and the copy or original may not necessarily have all the relevant settings to successfully resolve it (as 'nothing' ever does).